src/app/ReadClient.cpp

```diff
@@ -300,6 +300,10 @@ CHIP_ERROR ReadClient::OnMessageReceived(Messaging::ExchangeContext * apExchange
     {
         VerifyOrExit(apExchangeContext == mpExchangeCtx, err = CHIP_ERROR_INCORRECT_STATE);
         err = ProcessSubscribeResponse(std::move(aPayload));
+        // Forget the context as SUBSCRIBE RESPONSE is the last message in SUBSCRIBE transaction and
+        // ExchangeContext::HandleMessage automatically closes a context if no other messages need to
+        // be sent or received.
+        mpExchangeCtx = nullptr;
```

Comment:
Could you add a test to `TestReadInteraction` to case this case? You could keep it simple (something like `TestSubscribeRoundtrip` but without all the logic that sets things dirty. You'd just terminate the subscription after it's been established, and validate and check that the number of outstanding exchange contexts are 0).

Reply:
The number of outstanding exchange contexts will be 0 no matter what.  The question is whether this ReadClient has a dangling pointer or not....

Reply:
I added a test.